File: Strings/1120/1120.c
#include <stdio.h>
#include <string.h>


int main (){
    char digito, valor[200];
    int n, i, j;

    scanf("%c", &digito);
    scanf("%s", valor);

    while (digito != '0'){
        n = strlen(valor);
        for (i = 0; i < n; i++) {
            if (valor[i] == digito) {
                for (j = i; j < n; j++)
                    valor[j] = valor[j+1];
                valor[n-1] = 0;
                n--;
                i--;
            }
        }

        for (i = 0; i < n-1; i++) {
            if (valor[i] == '0') {
                for (j = i; j < n; j++)
                    valor[j] = valor[j+1];
                valor[n-1] = 0;
                n--;
                i--;
            }
            
            else
                break;
        }

        if (n == 0) {
            valor[0] = '0';
            valor[1] = '\0';
        }

        printf("%s\n", valor);

        scanf(" %c", &digito);
        scanf("%s", valor);
    }
    return 0;
}

File: Python/String/1120.py
# Problema 1120 - URI - Strings - Nível 5

# Inicializamos as variáveis, pois não existe Do While em python
d = ''
n = ''

while d != '0' and n != '0':
    # Leitura das entradas
    d, n = input().split(' ')

    # Nova variável para não perder o valor inicial de n
    result = n

    # Caso não seja o fim dos casos teste:
    if d != '0' and n != '0':

        # Removendo o digito d da string result
        result = result.replace(d,'')

        # Removendo zeros a esquerda
        while len(result) > 0 and result[0] == '0':
            # Metodo de substring que começa no segundo char e vai até o fim da string
            result = result[1:len(result)]

        # Caso a string tenha ficado vazia, atribuimos 0 a ela
        if len(result) == 0:
            result = '0'
        
        # Impressão do resultado
        print(result)